core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts

```diff
@@ -3168,6 +3559,394 @@ export class TopicManager extends BaseUser {
       throw newError;
     }
   }
+
+  /**
+   * Check if the save changes button is enabled or disabled in topic editor.
+   * @param {'enabled' | 'disabled'} status - The status to check.
+   */
+  async expectSaveChangesButtonInTopicEditorToBe(
+    status: 'enabled' | 'disabled'
+  ): Promise<void> {
+    await this.expectElementToBeVisible(saveTopicButton);
+    const buttonIsEnabled = await this.page.$eval(
+      saveTopicButton,
+      el => !(el as HTMLButtonElement).disabled
+    );
+    expect(buttonIsEnabled).toBe(status === 'enabled');
+  }
+
+  /**
+   * Navigates to the tab in the preview tab.
+   * @param {'Learn' | 'Practice' | 'Study'} tabName - The name of the tab.
+   */
+  async navigateToTabInPreview(
+    tabName: 'Learn' | 'Practice' | 'Study'
+  ): Promise<void> {
+    const xpath = `//*[contains(text(), '${tabName}')]`;
+    const tabElement = await this.page.waitForXPath(xpath);
+    if (!tabElement) {
+      throw new Error(`Tab ${tabName} not found.`);
+    }
+    await tabElement.click();
```

Comment:
can we use `this.clickOn()`?

Reply:
Currently, clickOn didn't worked properly with text content. However, I have updated it to use clickOn. I have also created a PR to fix the clickOn function.